### PR TITLE
Feature/php8

### PR DESCRIPTION
**Pull request by Baggerone**
One option for this is to review the code but tell me not to merge it until after ssp-base is tested against this feature branch.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,4 +4,4 @@ node_modules/
 composer.lock
 nbproject/
 .vagrant/
-local.env
+*local.env
```
Comment (forevermatt) [inline .gitignore:7]: Do you know if this still ignores "local.env", or does it only ignore "(something)local.env" now? In other words, does `*` also match the empty string?

File: development/idp4/saml20-idp-hosted.php [hunk 2]
```diff
@@ -9,6 +9,7 @@
     'authproc' => [
         10 => [
             'class' => 'mfa:Mfa',
+            'idpDomainName' => 'ssp-idp4.local',
```
Comment (forevermatt) [inline development/idp4/saml20-idp-hosted.php:12]: Did this addition end up making a difference?

File: docker-compose.yml [hunk 3]
```diff
@@ -140,7 +140,7 @@ services:
         MYSQL_PASSWORD: "pass"
 
   broker:
-    image: silintl/idp-id-broker:develop
+    image: silintl/idp-id-broker:latest
```
Comment (forevermatt) [inline docker-compose.yml:143]: I understand why some of the other dockage images were changed to the feature-branch tag, but why was this changed to `latest`?

**Comment by forevermatt** [comment]
> One option for this is to review the code but tell me not to merge it until after ssp-base is tested against this feature branch.

Yeah, waiting to merge this until ssp-base is updated to work with this code is probably a good idea. I don't remember all the ins-and-outs of these stutter-step updates of circular dependencies, though, so feel free to adjust as you need to. Ideally, `develop` will always contain code that could be released if we chose to, but I realize that's not always practical.

File: .gitignore (hunk 1 above)
Comment (Baggerone) [reply .gitignore:7]: Yep. It still ignores  "local.env".

File: development/idp4/saml20-idp-hosted.php (hunk 2 above)
Comment (Baggerone) [reply development/idp4/saml20-idp-hosted.php:12]: Yep. There was a change to module-silauth/src/auth/IdBroker.php that expected it.
```       $rpOrigin = 'https://' . $this->idpDomainName;```

**Review by forevermatt — APPROVED** [review]

**Comment by Baggerone** [comment]
I've tested manually this against the new `develop` version of ssp-base.
Unfortunately, any of the manual testing steps involving a user that has u2f/webauthn fail for me.

Do we wait and just test it in the appsdev IdP?

**Comment by Schparky** [comment]
> Do we wait and just test it in the appsdev IdP?

I'm fine with that. Seems like a better test than local anyway.

File: themes/material/mfa/low-on-backup-codes.php [hunk 4]
```diff
@@ -27,9 +27,10 @@
                     </h1>
                 </div>
 
+
                 <div class="mdl-card__title center" >
                     <p class="mdl-card__subtitle-text">
-                        <?= $this->t('{material:mfa:running_out_info}', ['{numBackupCodesRemaining}' => (int)$this->data['numBackupCodesRemaining']]) ?>
+                        <?= $this->t('{material:mfa:running_out_info}', ['{numBackupCodesRemaining}' => (string)(int)$this->data['numBackupCodesRemaining']]) ?>
```
Comment (forevermatt) [inline themes/material/mfa/low-on-backup-codes.php:33]: We might be able to remove both type casts here. It's currently casting to an `int`, then to ` string`, and I suspect it started as a string originally.
```suggestion
                        <?= $this->t('{material:mfa:running_out_info}', ['{numBackupCodesRemaining}' => $this->data['numBackupCodesRemaining']]) ?>
```

Comment (forevermatt) [reply themes/material/mfa/low-on-backup-codes.php:33]: Ahh, okay. Thanks.

**Review by forevermatt — APPROVED** [review]
